File: ConsoleApp7/Banc.cs
using System;
using System.Collections.Generic;
namespace Bankomat
{
    internal class Banc
    {
        private Dictionary<string, Client> _accounts = new Dictionary<string, Client>();

        public void AddAccount(Client client)
        {
            _accounts.Add(client.Login, client);
        }
        public bool FindLogin(string tryLogin, ref Client client)
        {
            return _accounts.TryGetValue(tryLogin, out client);
        }
    }
}

File: Bankomat/Banc.cpp
#include "Banc.h"

void Bankomat::Banc::AddAccount(Client client)
{
	_accounts[client.GetLogin()] = client;
}

bool Bankomat::Banc::FindLogin(std::string tryLogin, Client*& client)
{
	bool ans = _accounts.count(tryLogin);

	if(ans)
	client = (&_accounts.at(tryLogin));

	return ans;
}

bool Bankomat::Banc::FindLogin(std::string tryLogin)
{
	return _accounts.count(tryLogin);
}
